### PR TITLE
High-Level Design to reduce disk IO on SONiC switches

### PR DESCRIPTION
**Pull request by assrinivasan**
This PR is intended to introduce a High-level deisgn to reduce disk writes on SONiC switches

| Repo | PR Title |  State  |
|----------|:-------------:|------:|
| sonic-buildimage |Code optimizations to reduce disk writes on SONiC switches| [In Review](https://github.com/sonic-net/sonic-buildimage/pull/20640)

Signed-off-by: Ashwin Srinivasan [assrinivasan@microsoft.com](mailto:assrinivasan@microsoft.com)

### CODE REVIEW

File: doc/sonic-reduce-disk-io/sonic-reduce-disk-io.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+| SKU2  | Version_1  | 627.81      | 99.1            | 23.5             | 62.9           |
+|       | Version_3  | 729.14      | 43.3            | 8                | 77             |
+| SKU3  | Version_1  | 762.63      | 237.3           | 175.3            | 50             |
+|       | Version_2  | 591.26      | 160.1           | 94.4             | 52             |
+|       | Version_3  | 643.14      | 96              | 206              | 59             |
+
+##### **Table 4**
+
+Where:
+Optimization A: User space modifications, journaling is enabled and **/tmp is mounted to disk**
```
Comment (saiarcot895) [inline doc/sonic-reduce-disk-io/sonic-reduce-disk-io.md:107]: Use bullet points here for readability

File: doc/sonic-reduce-disk-io/sonic-reduce-disk-io.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+| HWSKU | OS Version | Unoptimized | Optimizations A |  Optimizations B | Optimization C |
+| ----- | ---------- | ----------- | --------------- | ---------------- | -------------- |
+| SKU1  | Version_1  | 872.89      | 147.1           | 135.9            | 66             |
+|       | Version_2  | 667.81      | 273.3           | 55.8             | 81             |
+|       | Version_3  | 719.52      | 133.2           | 87.5             | 70.4           |
+| SKU2  | Version_1  | 627.81      | 99.1            | 23.5             | 62.9           |
+|       | Version_3  | 729.14      | 43.3            | 8                | 77             |
+| SKU3  | Version_1  | 762.63      | 237.3           | 175.3            | 50             |
+|       | Version_2  | 591.26      | 160.1           | 94.4             | 52             |
+|       | Version_3  | 643.14      | 96              | 206              | 59             |
```
Comment (saiarcot895) [inline doc/sonic-reduce-disk-io/sonic-reduce-disk-io.md:102]: Do we know why the disk writes went up from 96MiB to 206MiB going from optimization A to B?

File: doc/sonic-reduce-disk-io/sonic-reduce-disk-io.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+![Reduction in disk writes as a percentage of writes in unoptimized SONiC images vs various optimization strategies](images/Reduction_Percentage.png)
+
+
+<br>
+<br>
+
+### Key Takeaways from Disk Write Comparisons Across Scenarios
+
+1. **Optimization A** consistently provides the highest reduction across most configurations, often exceeding 90%.
```
Comment (saiarcot895) [inline doc/sonic-reduce-disk-io/sonic-reduce-disk-io.md:136]: Looking at the percentages above, doesn't optimization B reduce disk writes further than optimization A in all but one scenario?

File: doc/sonic-reduce-disk-io/sonic-reduce-disk-io.md (hunk 2 above)
Comment (assrinivasan) [reply doc/sonic-reduce-disk-io/sonic-reduce-disk-io.md:102]: I am not entirely sure. The caveat is that this was derived from 1-2 datapoints as opposed to 100 as was the case with the unoptimized scenario. That leaves these numbers open to a fair amount of variability as is seen the reduction % values across SKUs and OS versions (59% - 98%). I will condense this bar chart down to a mean value of all 3 strategies across SKUs/OSes to guard against trying to derive meaning from any single data point.

File: doc/sonic-reduce-disk-io/sonic-reduce-disk-io.md (hunk 3 above)
Comment (assrinivasan) [reply doc/sonic-reduce-disk-io/sonic-reduce-disk-io.md:136]: Hey good catch, that was supposed to say Optimization A. Will change.

File: doc/sonic-reduce-disk-io/sonic-reduce-disk-io.md (hunk 1 above)
Comment (assrinivasan) [reply doc/sonic-reduce-disk-io/sonic-reduce-disk-io.md:107]: Thanks for the catch - will add bullet points in upcoming commit.

File: doc/sonic-reduce-disk-io/sonic-reduce-disk-io.md (hunk 2 above)
Comment (assrinivasan) [reply doc/sonic-reduce-disk-io/sonic-reduce-disk-io.md:111]: Done in the latest commit, lmk your thoughts.

Comment (prgeor) [reply doc/sonic-reduce-disk-io/sonic-reduce-disk-io.md:111]: image
sorry its not clear 96 -> 206 MB with optimization B

**Review by saiarcot895 — APPROVED** [review]

**Review by prgeor — APPROVED** [review]

**Pull request merged**